utils/pipeline_utils.py

```
@@ -387,6 +414,34 @@ def optimize(pipeline, args):
     # switch memory layout to channels_last
     if not args.disable_channels_last:
         pipeline.vae.to(memory_format=torch.channels_last)
+    
+    # cache-dit: DBCache configs
+    if args.enable_cache_dit:
+        try:
+            from cache_dit.cache_factory import apply_cache_on_pipe, CacheType
+            # docs: https://github.com/vipshop/cache-dit
+            cache_options = {
+                "cache_type": CacheType.DBCache,
+                "warmup_steps": args.warmup_steps,
+                "max_cached_steps": args.max_cached_steps,
+                "Fn_compute_blocks": args.Fn_compute_blocks,
+                "Bn_compute_blocks": args.Bn_compute_blocks,
+                "residual_diff_threshold": args.residual_diff_threshold,
+                # TaylorSeer options
+                "enable_taylorseer": args.enable_taylorseer,
+                "enable_encoder_taylorseer": args.enable_taylorseer,
+                # Taylorseer cache type cache be hidden_states or residual
+                "taylorseer_cache_type": "residual",
+                "taylorseer_kwargs": {
+                    "n_derivatives": 2,
+                },
+            }
+            apply_cache_on_pipe(pipeline, **cache_options)
+        except ImportError:
+            print(
+                "Please install cache-dit via 'pip install -U cache-dit'"
+            )
+            pass
```

nit: I think it's best to error out entirely instead of swallowing this exception

---

done